Brokerages/Bitfinex/BitfinexBrokerageFactory.cs

```diff
@@ -0,0 +1,108 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using QuantConnect.Util;
+using QuantConnect.Interfaces;
+using QuantConnect.Configuration;
+using TradingApi.Bitfinex;
+
+namespace QuantConnect.Brokerages.Bitfinex
+{
+
+    /// <summary>
+    /// Factory method to create Bitfinex Websockets brokerage
+    /// </summary>
+    public class BitfinexBrokerageFactory : BrokerageFactory
+    {
+
+        /// <summary>
+        /// Factory constructor
+        /// </summary>
+        public BitfinexBrokerageFactory()
+            : base(typeof(BitfinexBrokerage))
+        {
+        }
+
+        /// <summary>
+        /// Not required
+        /// </summary>
+        public override void Dispose()
+        {
+
+        }
+
+
+        /// <summary>
+        /// provides brokerage connection data
+        /// </summary>
+        public override Dictionary<string, string> BrokerageData
+        {
+            get
+            {
+                return new Dictionary<string, string>
+                {
+                };
+            }
+        }
+
+        /// <summary>
+        /// The brokerage model
+        /// </summary>
+        public override IBrokerageModel BrokerageModel
+        {
+            get { return new BitfinexBrokerageModel(); }
+        }
+
+        /// <summary>
+        /// Create the Brokerage instance
+        /// </summary>
+        /// <param name="job"></param>
+        /// <param name="algorithm"></param>
+        /// <returns></returns>
+        public override Interfaces.IBrokerage CreateBrokerage(Packets.LiveNodePacket job, Interfaces.IAlgorithm algorithm)
+        {
+
+            string apiSecret = Config.Get("bitfinex-api-secret");
+            string apiKey = Config.Get("bitfinex-api-key");
+            string wallet = Config.Get("bitfinex-wallet");
+            //it's desirable to throw an exception here when failing parse
+            decimal scaleFactor = decimal.Parse(Config.Get("bitfinex-scale-factor", "1"));
```

Comment:
All of these config values need to come from the `job.BrokerageData` dictionary. You can move these config gets into the `BrokerageData` property above. Have a look at the [InteractiveBrokersBrokerageFactory](https://github.com/QuantConnect/Lean/blob/master/Brokerages/InteractiveBrokers/InteractiveBrokersBrokerageFactory.cs#L55) for an example.